src/interpolations.jl

```diff
@@ -1540,6 +1538,42 @@ function shape_gradient_and_value(ipv::VectorizedInterpolation{vdim, shape}, ξ:
     return SMatrix(grad), val
 end
 
+# vdim == refdim
+function shape_hessian_gradient_and_value(ipv::VectorizedInterpolation{dim, shape}, ξ::Vec{dim}, I::Int) where {dim, shape <: AbstractRefShape{dim}}
+    return invoke(shape_hessian_gradient_and_value, Tuple{Interpolation, Vec, Int}, ipv, ξ, I)
+end
+# vdim != refdim
+function shape_hessian_gradient_and_value(ipv::VectorizedInterpolation{vdim, shape}, ξ::V, I::Int) where {vdim, refdim, shape <: AbstractRefShape{refdim}, T, V <: Vec{refdim, T}}
+    _shape_hessian_gradient_and_value_static_array(ipv, ξ, I)
+end
+function _shape_hessian_gradient_and_value_static_array(ipv::VectorizedInterpolation{vdim, shape}, ξ::V, I::Int) where {vdim, refdim, shape <: AbstractRefShape{refdim}, T, V <: Vec{refdim, T}}
+    # Load with dual numbers and compute the value
+    f = x -> shape_value(ipv, x, I)
+    ξd =  Tensors._load(Tensors._load(ξ, Tensors.Tag(f, V)), Tensors.Tag(f, V))
+    value_hess = f(ξd)
+    # Extract the value and gradient
+    val = Vec{vdim, T}(i -> Tensors.value(Tensors.value(value_hess[i])))
+    grad = zero(MMatrix{vdim, refdim, T})
+    hess = zero(MArray{Tuple{vdim, refdim, refdim}, T})
+    for (i, vi) in pairs(value_hess)
+        hess_values = Tensors.value(vi)
+        
+        hess_values_partials = Tensors.partials(hess_values)
+        for (k, pk) in pairs(hess_values_partials)
+            grad[i, k] = pk
+        end
+    
+        hess_partials = Tensors.partials(vi)
+        for (j, partial_j) in pairs(hess_partials)
+            hess_partials_partials = Tensors.partials(partial_j)
+            for (k, pk) in pairs(hess_partials_partials)
+                hess[i, j, k] = pk
+            end
+        end
+    end
+    return SArray(hess), SMatrix(grad), val
+end
```

Comment:
Couldn't really come up with a similar simplification here as in #947, since `ForwardDiff.jacobian` will stack the output in a strange way, so I suggest leaving this as it is for now while waiting for https://github.com/Ferrite-FEM/Tensors.jl/pull/188.

A future optimization would be to special-case for `VectorizedInterpolation` and only calculate the values and gradients wrt. the "active" value, but that is a separate issue.